desktop-widgets/diveplanner.ui

```diff
@@ -163,73 +128,141 @@
-          <number>689</number>
+          <number>-100</number>
          </property>
          <property name="maximum">
-          <number>1100</number>
+          <number>3000</number>
+         </property>
+         <property name="singleStep">
+          <number>10</number>
          </property>
         </widget>
        </item>
-       <item row="3" column="0">
-        <widget class="QSpinBox" name="atmHeight">
+       <item row="3" column="1">
+        <widget class="QSpinBox" name="ATMPressure">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="suffix">
-          <string>m</string>
+          <string>mbar</string>
          </property>
          <property name="minimum">
-          <number>-100</number>
+          <number>689</number>
          </property>
          <property name="maximum">
-          <number>3000</number>
-         </property>
-         <property name="singleStep">
-          <number>10</number>
+          <number>1100</number>
          </property>
         </widget>
        </item>
        <item row="3" column="2">
-        <widget class="QDoubleSpinBox" name="salinity">
+        <widget class="QComboBox" name="waterType">
+         <item>
+          <property name="text">
+           <string>Fresh water</string>
+          </property>
+         </item>
+         <item>
+          <property name="text">
+           <string>Sea water</string>
+          </property>
+         </item>
+         <item>
+          <property name="text">
+           <string>EN13319</string>
+          </property>
+         </item>
+         <item>
+          <property name="text">
+           <string>Custom</string>
+          </property>
+         </item>
+        </widget>
+       </item>
+       <item row="3" column="3">
+        <widget class="QDoubleSpinBox" name="customSalinity">
+         <property name="enabled">
+          <bool>false</bool>
+         </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
+         <property name="maximumSize">
+          <size>
+           <width>90</width>
+           <height>16777215</height>
+          </size>
+         </property>
+         <property name="toolTip">
+          <string extracomment="Custom water density"/>
+         </property>
          <property name="suffix">
           <string>kg/ℓ</string>
          </property>
          <property name="minimum">
-          <double>1.000000000000000</double>
+          <double>0.990000000000000</double>
          </property>
          <property name="maximum">
-          <double>1.050000000000000</double>
+          <double>1.300000000000000</double>
```

Comment:
1.3 seems, err, high :-)

Reply:
My profound Wikipedia knowledge says that the dead sea has about 1.24kg/l and that some extremely salty lakes have even higher values! 0.99kg/l is also an extreme value that only occurs in fresh water at >40°C. So this might even be more unrealistic. However, I believe that these boundaries won't cause any trouble and I don't want to disallow values that are theoretically possible..

Reply:
wow. thanks for the education on that. I realized that the dead sea was very salty, but had no idea it was THAT salty. So this is "good as merged"